src/tests/ftest/soak/soak.py

```diff
@@ -0,0 +1,338 @@
+            # queue it up to run and register a callback to retrieve results
+            job_id = slurm_utils.run_slurm_script(script1)
+            slurm_utils.register_for_job_results(job_id, job_done, maxwait=3600)
+
+            # queue up a second job
+            s1_job2_name = self.params.get("name", '/run/job2/')
+            s1_job2_nodes = self.params.get("nodes", '/run/job2/')
+            s1_job2_processes = self.params.get("process_per_node",
+                                                '/run/job2/')
+            s1_job2_spec = self.params.get("jobspec", '/run/job2/')
+            cmd = [self.get_ior_job(s1_job2_spec, s1_job2_processes,
+                                    s1_job2_name)]
+
+            output = os.path.join(self.tmpdir, s1_job2_name + "_results.out")
+            script2 = slurm_utils.write_slurm_script(self.tmpdir, s1_job2_name,
+                                                    output,
+                                                    int(s1_job2_nodes), cmd)
+            id2 = slurm_utils.run_slurm_script(script2)
+            slurm_utils.register_for_job_results(id2, job_done, maxwait=3600)
+
+            # wait for all the jobs to finish
+            global _soak_results
+            while len(_soak_results) < 2:
+                time.sleep(10)
+
+            for job, result in _soak_results.iteritems():
+                if result != "COMPLETED":
+                    self.fail("Soak job: {} didn't complete as expected: {}".
+                              format(job, result))
+
+        except (DaosApiError, IorUtils.IorFailed) as error:
+            self.fail("<Soak Test 1 Failed>\n {}".format(error))
+        finally:
+            try:
+                os.remove(script1)
+            except:
+                pass
+            try:
+                os.remove(script2)
+            except:
+                pass
+
+    def test_soak_2(self):
+        """
+        Test ID: DAOS-2192
+        Test Description: This test verifies that a dmg script can be submitted.
+        :avocado: tags=soak2
+        """
+
+        script = None
+        try:
+            dmgcmds = DmgUtils.get_dmg_script("dmg1", self.params,
+                                              self.basepath,
+                                              self.workdir)
+
+            s2_job1_name = self.params.get("name", '/run/job3/')
+            s2_job1_nodes = self.params.get("nodes", '/run/job3/')
+
+            output = os.path.join(self.tmpdir, s2_job1_name + "_results.out")
+
+            script = slurm_utils.write_slurm_script(self.tmpdir, s2_job1_name,
+                                                   output,
+                                                   s2_job1_nodes, dmgcmds)
+            job_id = slurm_utils.run_slurm_script(script)
+            slurm_utils.register_for_job_results(job_id, job_done, maxwait=3600)
+
+            # wait for all the jobs to finish
+            global _soak_results
```

Comment:
(pylint-global-variable-not-assigned) Using global for '_soak_results' but no assignment is done

(pylint-invalid-name) Invalid constant name "_soak_results"